core/src/prediction/collector/DefaultPredictionCollector.cpp

```diff
@@ -35,6 +35,7 @@ std::vector<Prediction> DefaultPredictionCollector::collect_predictions(
     const std::vector<std::vector<bool>>& valid_trees_by_sample,
     bool estimate_variance,
     bool estimate_error) const {
+  // Note: estimate_error is not necessary for the DefaultPredictionStrategy and is unused here.
```

Comment:
Small comment, I think it'd be cleaner + more visible to add this as a method comment on the interface file `DefaultPredictionCollector.h`